### PR TITLE
refactor: simplify event index

### PR DESCRIPTION
**Pull request by igboyes**
* Remove duplicate, conflicting, or unused fixtures.
* Fix conflicting repo fixture paths.
* Remove unused repo and event index methods.

### CODE REVIEW

File: tests/test_add_bulk.py
Comment (sygao) [inline tests/test_add_bulk.py:1]: We're still to need tests that simulate building and and querying a reasonably-sized repo from the interface. Ferreting out the more difficult bugs in `Snapshotter` and `EventIndex` will probably require larger tests of this nature.

File: virtool_cli/ref/event_index_cache.py [hunk 1]
```diff
@@ -1,126 +1,84 @@
...
+        return event_index
 
-        try:
-            with open(self.path / f"{otu_id}.json", "wb") as f:
-                f.write(orjson.dumps(dataclasses.asdict(index_data)))
-        finally:
-            logger.debug(
-                "OTU event index updated", otu_id=str(otu_id), events=event_list
-            )
+    def get(self, otu_id: UUID) -> EventIndexItem | None:
```
Comment (sygao) [inline virtool_cli/ref/event_index_cache.py:65]: `.get()` is a little too stripped down for my comfort. `.load()` appears to return an entire dictionary, after all.

I think we should still reference OTU in the method name.

**Review by sygao — CHANGES_REQUESTED** [review]

File: virtool_cli/ref/event_index_cache.py (hunk 1 above)
Comment (igboyes) [reply virtool_cli/ref/event_index_cache.py:65]: I'm not sure what you mean. What is it missing?

File: tests/test_add_bulk.py
Comment (igboyes) [reply tests/test_add_bulk.py:1]: I agree, but I don't think the rest of the code works well enough right now to do a good job of it.

This test module did not include the complete isolate information and it turned out it was glossing over ordering issues and isolates being binned inconsistently and incorrectly.

Once the units are rock solid, let's do it.

File: virtool_cli/ref/event_index_cache.py [hunk 2]
```diff
@@ -1,126 +1,84 @@
...
+class EventIndexItem:
+    at_event: int
     """The latest event ID when this cache index was last updated"""
 
-    events: list[int]
+    event_ids: list[int]
     """A list of event IDs"""
 
-    @model_validator(mode="after")
-    def sort_events(self):
```
[the remark this answers is not in the document]
Comment (igboyes) [reply virtool_cli/ref/event_index_cache.py:28]: I've added a test to make sure unsorted event IDs are sorted on write.

File: virtool_cli/ref/event_index_cache.py (hunk 1 above)
Comment (sygao) [reply virtool_cli/ref/event_index_cache.py:69]: `.get()`/`.set()` deal with single OTUs, `.load()` deals with a batch of them. The scale of the operation isn't adequately implied in the method names.

File: virtool_cli/ref/event_index_cache.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sygao) [reply virtool_cli/ref/event_index_cache.py:28]: I put this in to deal with some bugs that popped up while I was developing the thing, but you're right. I've probably put in too many hacky failsafes as stopgaps.

**Pull request merged**